### PR TITLE
aws-sam-cli: 0.16.1 -> 0.19.0

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
Saw a weird error while reviewing another pr

Actions:
- python2 -> python3
- version bump
- made python2 only libs optional
- added longDescription

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @andreabedini 
@FRidh  because it's most a python package.

### CODE REVIEW

File: pkgs/development/tools/aws-sam-cli/default.nix [hunk 1]
```diff
@@ -49,22 +57,25 @@ buildPythonApplication rec {
...
   '';
 
   meta = with lib; {
     homepage = https://github.com/awslabs/aws-sam-cli;
     description = "CLI tool for local development and testing of Serverless applications";
     license = licenses.asl20;
     maintainers = with maintainers; [ andreabedini dhkl ];
+
+    longDescription = ''
+      The AWS Serverless Application Model (SAM) is an open-source framework for building serverless applications. It provides shorthand syntax to express functions, APIs, databases, and event source mappings. With just a few lines of configuration, you can define the application you want and model it.
```
Comment (worldofpeace) [inline pkgs/development/tools/aws-sam-cli/default.nix]: please use line breaks

File: pkgs/development/tools/aws-sam-cli/default.nix [hunk 2]
```diff
@@ -13,6 +13,15 @@ let
         };
       });
 
+      requests = super.requests.overridePythonAttrs (oldAttrs: rec {
+        version = "2.22.0";
```
Comment (worldofpeace) [inline pkgs/development/tools/aws-sam-cli/default.nix]: Don't we have this in nixpkgs?
```
 nix eval -f . python3.pkgs.requests.version
"2.22.0"
```

File: pkgs/development/tools/aws-sam-cli/default.nix [hunk 3]
```diff
@@ -13,6 +13,15 @@ let
...
+        src = oldAttrs.src.override {
+          inherit version;
+          sha256 = "1d5ybh11jr5sm7xp6mz8fyc7vrp4syifds91m7sj60xalal0gq0i";
+        };
+      });
+
+
       aws-sam-translator = super.aws-sam-translator.overridePythonAttrs (oldAttrs: rec {
```
Comment (worldofpeace) [inline pkgs/development/tools/aws-sam-cli/default.nix]: This is interesting.

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -647,7 +647,9 @@ in
 
   aws_shell = pythonPackages.callPackage ../tools/admin/aws_shell { };
 
-  aws-sam-cli = callPackage ../development/tools/aws-sam-cli { };
+  aws-sam-cli = callPackage ../development/tools/aws-sam-cli {
+    python = python3;
```
Comment (worldofpeace) [inline pkgs/top-level/all-packages.nix]: can we just use `python3` in the expression?

**Review by worldofpeace — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/aws-sam-cli/default.nix (hunk 2 above)
Comment (jonringer) [reply pkgs/development/tools/aws-sam-cli/default.nix]: currently, yes

File: pkgs/top-level/all-packages.nix (hunk 4 above)
Comment (jonringer) [reply pkgs/top-level/all-packages.nix]: sure

File: pkgs/development/tools/aws-sam-cli/default.nix (hunk 2 above)
Comment (jonringer) [reply pkgs/development/tools/aws-sam-cli/default.nix]: must have been a recent change, it was 2.21 when i wrote the pr

**Comment by jonringer** [comment]
Rebased on top of master, and applied changes.

**Comment by jonringer** [comment]
@GrahamcOfBorg build aws-sam-cli

**Comment by jonringer** [comment]
@worldofpeace @marsam rebased on top of master, applied changes 6 weeks ago :)

**Comment by worldofpeace** [comment]
@jonringer We'll see about integrating. And commit bit soon :rofl:

File: pkgs/development/tools/aws-sam-cli/default.nix [hunk 5]
```diff
@@ -26,14 +35,13 @@ let
 in
 
 with py.pkgs;
-
 buildPythonApplication rec {
   pname = "aws-sam-cli";
-  version = "0.16.1";
+  version = "0.19.0";
```
Comment (worldofpeace) [inline pkgs/development/tools/aws-sam-cli/default.nix]: https://github.com/awslabs/aws-sam-cli/tree/v0.22.0

It's unfortunate this was open so long for the update to be out of date.
But a successful nudge helped with at least one thing :smile:

Comment (jonringer) [reply pkgs/development/tools/aws-sam-cli/default.nix]: sigh..

Comment (jonringer) [reply pkgs/development/tools/aws-sam-cli/default.nix]: busting my chops... lol :rofl:

**Comment by jonringer** [comment]
@GrahamcOfBorg build aws-sam-cli

**Comment by jonringer** [comment]
```
[3 built, 1 copied (0.1 MiB), 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/66125
5 package were build:
aws-sam-cli python27Packages.aws-lambda-builders python27Packages.serverlessrepo python37Packages.aws-lambda-builders python37Packages.serverlessrepo
```

**Comment by jonringer** [comment]
master is now at 0.22.0, closing

**Pull request closed without merging**